src/date-utils.es6

```diff
@@ -156,21 +156,86 @@ const DateUtils = {
     return moment(date)
   },
 
+
+  /**
+   * Return a formatting string for displaying time
+   *
+   * @param {Date} opts - Object with different properties for customising output
+   * @return {String} The format string based on syntax used by Moment.js
+   *
+   * seconds, upperCase and timeZone are the supported extra options to the format string.
+   * Checks whether or not to use 24 hour time format.
+   */
   getTimeFormat(opts) {
     const use24HourClock = NylasEnv.config.get('core.workspace.use24HourClock')
     let timeFormat = use24HourClock ? "HH:mm" : "h:mm"
+
     if (opts && opts.seconds) {
       timeFormat += ":ss"
     }
-    if (!use24HourClock && opts && opts.upperCase) {
-      timeFormat += " A"
+
+    // Append meridian if not using 24 hour clock
+    if (!use24HourClock) {
+      if (opts && opts.upperCase) {
+        timeFormat += " A"
+      } else {
+        timeFormat += " a"
+      }
     }
+
     if (opts && opts.timeZone) {
       timeFormat += " z"
     }
 
     return timeFormat
   },
+
+
+  /**
+   * Return a short format date/time
+   *
+   * @param {Date} datetime - Timestamp
+   * @return {String} Formated date/time
+   *
+   * The returned date/time format depends on how long ago the timestamp is.
+   */
+  shortTimeString(datetime) {
+    const diff = moment().diff(datetime, 'days', true)
+    let format = null
+
+    if (diff <= 1) {
+      // Time if less than 1 day old
+      format = DateUtils.getTimeFormat(null)
+    } else if (diff > 1 && diff <= 365) {
+      // Month and day up to 1 year old
+      format = "MMM D"
+    } else {
+      // Month, day and year if over a year old
+      format = "MMM D YYYY"
+    }
+
+    return moment(datetime).format(format)
+  },
+
+
+  /**
+   * Return a long format date/time
+   *
+   * @param {Date} datetime - Timestamp
+   * @return {String} Formated date/time
+   */
+  fullTimeString(datetime) {
+    const tz = moment.tz.guess()
```

Comment:
I think that moment.tz.guess does a lot of work internally when the timezone isn't easy to determine. (It does a bunch of date math with common dates to identify what the time zone must be.) Since we call this function really often (as the user is scrolling in the thread list) and performance is super important, I'd pull this back out into global scope at the top of the file.

That /does/ mean that the user changing timezones won't be detected. We might want to create a getter method for it, and add a bit of logic to re-compute it at most every five minutes, or something like that if having it update is important. (Might want to make it a separate PR to avoid expanding the scope of this one too much.)